_posts/2021-01-04-Designing-REST-services.md

```diff
@@ -0,0 +1,331 @@
+Web services are software components that are developed to expose business capabilities. 
+To do so, they are comprised of the following elements (available through the provided contract):
+* Operations: A set of exportable operation signatures that can be accessed by the consumer of the service.
+* Document Schemas: A definition of the data types that can be exchanged with the service through the provided interfaces.
+* Non-Functional Specifications: 
+    * Conversation Specifics: Indication on how information can be exchanged with the services, such as request-response or fire-and-forget.
+    * Quality-of-Service Characteristics (QoS): Indicators for QoS characteristics such as availability, latency and throughput.
+    * Policy Specifics: Requirements specifications for how to interact with the service. These are stipulations on for example security and transactional contexts.
+
+The first five chapters of the book deal mainly in building a transformative journey for an organization to shift to using REST services according to the Richardson’s maturity model. 
+The model starts from a very basic implementation not following any of the REST standards up to a full implementation of services with the principle of “Hypermedia As The Engine Of Application State” (HATEOS). 
+The remaining chapters deal with how to address any non-functional specifications these services must cover. 
+In these remaining chapters there is a brief intermezzo where the authors delve into the ATOM format as a substitute for transferring data and a means for event driven service design. 
+As these chapters are a bit the odd ones out, this synopsis will not delve into them too deeply.
+
+{:refdef: style="text-align: center;"}
+<img src="{{ '/img/2021-01-04-Designing-REST-services/maturitymodel.png' | prepend: site.baseurl }}" alt="Workspace" class="image" style="margin:0px auto; max-width:100%">
+{: refdef}
+{:refdef: style="text-align: center;"}
+Richardson’s Maturity Model
+{: refdef}
+
+##	Level 1 - Resources
+
+In order to clarify the success factors of the WWW as a platform for services on a global scale, the book sketches a high-level overview of the architecture behind it, and the salient points for the REST architecture. 
+Formalized by the World Wide Web Consortium (W3C), the specification can be found [here](https://www.w3.org/TR/webarch/){:target="_blank" rel="noopener noreferrer"}. 
+Services are exposing resources through Uniform Resource Identifiers (URI). 
+Their relation is many-to-one: A resource can be identified by more than one URI, but any URI will only point to one resource, making it addressable and accessible for manipulation via HTTP.
+This endpoint will follow a specific structure: &lt;protocol&gt;://&lt;host&gt;/&lt;resource&gt;/&lt;identifier&gt; with the identifier piece being optional. 
+To fetch consultant Peter De Kinder from the service would look something like this:
+```
+    https://www.bestitpeople.com/consultant/peterdekinder
```

Comment:
It doesn't say new, it says newer. But I will add quotes around newer to make this more specific :)

Reply:
The reason all the URLs use singular is because the book puts them all in singular. See the sidebar that I added on the choice between singular and plural.